caffe2/CMakeLists.txt

```diff
@@ -337,6 +337,10 @@ if(NOT INTERN_BUILD_MOBILE OR NOT BUILD_CAFFE2_MOBILE)
     "${TOOLS_PATH}/shared/_utils_internal.py"
     COPYONLY)
 
+  # Generate header with version info
+  configure_file("${TORCH_SRC_DIR}/csrc/api/include/torch/version.h.in"
```

Comment:
You'll probably need to add `header_template_rule` to BUILD.bazel similar to 
https://github.com/pytorch/pytorch/blob/12ee7b61e75caa0c57b53ee1627e3075686d2510/BUILD.bazel#L41-L48

Reply:
> You'll probably need to add `header_template_rule` to BUILD.bazel similar to
> https://github.com/pytorch/pytorch/blob/12ee7b61e75caa0c57b53ee1627e3075686d2510/BUILD.bazel#L41-L48

@malfet 
It seems to me that we did not setup the version macros in BUILD.bazel (correct me if I am wrong)
I find this https://github.com/pytorch/pytorch/blob/master/BUILD.bazel#L786, but is it hard coded to me.

Reply:
@glaringlee we do not ship any binaries from bazel build, so version numbers can be stale.

Reply:
@malfet thanks for comment.
@jbschlosser per @malfet's comment above, it is fine that we just use a hard coded version for TORCH_VERSION_XXXX in bzl, but if you are in the middle of looking at bzl action to parse version.txt and it is not too time consuming, it is also great to let bzl support versioning. You can do either way, let me know when you think this is ready.

Reply:
Great- thanks for looking into it! I have it working for now hard-coded, but I want to attempt the proper solution. Will let you know when I'm ready.

Reply:
sure, thank you!

Reply:
@glaringlee Well, I now have the version header being generated from `version.txt` in Bazel. Since Bazel cannot read file contents at analysis time in a way that affects the build (see [here](https://stackoverflow.com/questions/56007307/read-local-file-content-in-bazel-build-system)), I could not use `header_template_rule` and had to switch to a `genrule` that launches a python script that does manual template expansion. It works but it's ugly. Perhaps it's best to keep the hard-coded version for now and solve this properly later on.

Reply:
@malfet This PR now uses `genrule` to parse version.txt in bazel build to support torch version number in c++.  The code introduced a helper .py script to parse version.txt and generate the version header file, it seems fine to me. Can you advise?